privacyidea/lib/tokens/papertoken.py

```diff
@@ -132,11 +132,15 @@ def get_class_info(key=None, ret='all'):
         return ret
 
     def update(self, param, reset_failcount=True):
-        if "otpkey" not in param:
+        if "otpkey" not in param and "verify" not in param:
             param["genkey"] = 1
         HotpTokenClass.update(self, param, reset_failcount=reset_failcount)
         papertoken_count = int(param.get("papertoken_count", DEFAULT_COUNT))
         # Now we calculate all the OTP values and add them to the
-        # init_details. Thus they will be returned by token/init.
-        otps = self.get_multi_otp(count=papertoken_count)
-        self.add_init_details("otps", otps[2].get("otp", {}))
+        # init_details. Thus, they will be returned by token/init.
+        # But only if this is not in verify state.
+        # TODO: check if can finish the request after verifying the token in the
```

Comment:
Just one note: You are adding todos in the code a lot for aspects, that might be nice one day. I would not add todos for things, which we most probably will not/never do. ;-)

Reply:
Yeah, i kind of like TODOs :-)
I think for notes that something is missing (like specific tests) this can be useful to have it in the code.